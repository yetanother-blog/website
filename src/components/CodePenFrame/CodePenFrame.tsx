import React from 'react';

interface CodePenFrameProps {
    url: string;
    title: string;
    height?: number;
}

export const CodePenFrame: React.FC<CodePenFrameProps> = ({url, height = 500, title}) => {
    return <div dangerouslySetInnerHTML={{ __html: `
        <iframe 
            height="${height}"
            style="width: 100%;"
            scrolling="no"
            title="${title}"
            src="${url}?height=${height}&theme-id=light&default-tab=result"
            frameborder="no"
            loading="lazy"
            allowtransparency="true"
            allowfullscreen="true" /> 
    `}}
    />;
}
