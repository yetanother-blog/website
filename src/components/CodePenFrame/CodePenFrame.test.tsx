import React from 'react'
import { render } from '@testing-library/react'
import {CodePenFrame} from './CodePenFrame'

describe('CodePenFrame component', () => {
    it('should render', () => {
        const { container } = render(<CodePenFrame url="https://some-code-pen.url" title="My ðŸ˜Ž CodePen"/>);
        expect(container).toMatchSnapshot();
    });
});