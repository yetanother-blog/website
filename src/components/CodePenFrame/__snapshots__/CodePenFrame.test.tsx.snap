// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodePenFrame component should render 1`] = `
<div>
  <div>
    
        
    <iframe
      allowfullscreen="true"
      allowtransparency="true"
      frameborder="no"
      height="500"
      loading="lazy"
      scrolling="no"
      src="https://some-code-pen.url?height=500&theme-id=light&default-tab=result"
      style="width: 100%;"
      title="My ðŸ˜Ž CodePen"
    >
       
    
    </iframe>
  </div>
</div>
`;
